packages/components/src/templates/next/components/complex/Image/Image.tsx

```diff
@@ -1,26 +1,93 @@
+import type { PropsWithChildren } from "react"
+
 import type { ImageProps } from "~/interfaces"
+import { tv } from "~/lib/tv"
+import { isExternalUrl } from "~/utils"
+import { Link } from "../../internal/Link"
+
+const createImageStyles = tv({
+  slots: {
+    container: "mt-0 [&:not(:first-child)]:mt-7",
+    caption:
+      "overflow-wrap break-word prose-label-sm-medium mt-2 max-w-[100ch] text-base-content-subtle lg:mx-auto lg:text-center",
+    image: "mx-auto h-auto max-w-full rounded",
+  },
+  variants: {
+    size: {
+      small: {
+        image: "min-w-full xs:min-w-[67%] md:min-w-[33%] lg:min-w-0",
+      },
+      medium: {
+        image: "min-w-full xs:min-w-[33%] md:min-w-0",
+      },
+      large: { image: "min-w-0" },
+    },
+  },
+})
+const compoundStyles = createImageStyles()
+
+// Enforce a certain minimum width of image if screen size is small
+const getWidthSize = (width: ImageProps["width"]) => {
+  if (width === undefined || width >= 66) {
+    return "large"
+  } else if (width >= 40) {
+    return "medium"
+  } else {
+    return "small"
+  }
+}
```

Comment:
note to self: to check with design @sehyunidaaa on what the outcome is regarding the sizing, and whether we want to do that change right now or in a future PR